### ISSUE TITLE
ESPNet 1 pythorch Snapshot for every epoch

### ISSUE THREAD

**Issue by szmmm** [espnet/espnet#2676]
Hello! I realised ESPNet produces a snapshot for each one of the epochs. Is there a setting where I could control the amount of snapshot the model makes, for example every 3 or 5 epochs? In ESPNet2 I think it automatically deletes some of the snapshots to save disk space, so I wonder if ESPNet does something like that too. Thank you!

**Comment by Fhrozen** [comment]
The easy way is for you to modify the option of every epoch at https://github.com/espnet/espnet/blob/5d97a488c52cd303993064e147fb3a2be20337a5/espnet/asr/pytorch_backend/asr.py#L794-L795 and change it at n-epochs. 
If you want to keep only the best n-last epochs then you will need to add an extension to delete previous snapshots with low acc or high loss. Probably, similar to this https://github.com/chainer/chainer/blob/f747f478beb42795b100ff666d83fd48d7b5495d/chainer/training/extensions/_snapshot.py#L383-L396

**Comment by szmmm** [comment]
> The easy way is for you to modify the option of every epoch at
> 
> https://github.com/espnet/espnet/blob/5d97a488c52cd303993064e147fb3a2be20337a5/espnet/asr/pytorch_backend/asr.py#L794-L795
> 
> and change it at n-epochs.
> If you want to keep only the best n-last epochs then you will need to add an extension to delete previous snapshots with low acc or high loss. Probably, similar to this https://github.com/chainer/chainer/blob/f747f478beb42795b100ff666d83fd48d7b5495d/chainer/training/extensions/_snapshot.py#L383-L396

Thank you very much!
